lib/child_process.js

```diff
@@ -276,31 +275,34 @@ ObjectDefineProperty(exec, promisify.custom, {
  *   ) => any} [callback]
  * @returns {ChildProcess}
  */
-function execFile(file /* , args, options, callback */) {
-  let args = [];
-  let callback;
-  let options;
-
-  // Parse the optional positional parameters.
-  let pos = 1;
-  if (pos < arguments.length && ArrayIsArray(arguments[pos])) {
-    args = arguments[pos++];
-  } else if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
-  }
-
-  if (pos < arguments.length && typeof arguments[pos] === 'object') {
-    options = arguments[pos++];
-  } else if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
+function execFile(file, args = [], options, callback) {
+  if (args == null) {
+    args = [];
+  } else if (typeof args === 'object') {
+    if (!ArrayIsArray(args)) {
+      callback = options;
+      options = args;
+      args = [];
+    }
+  } else if (typeof args === 'function') {
+    callback = args;
+    options = {};
+    args = [];
+  } else {
+    throw new ERR_INVALID_ARG_VALUE('args', args);
   }
 
-  if (pos < arguments.length && typeof arguments[pos] === 'function') {
-    callback = arguments[pos++];
+  if (options == null) {
+    options = {};
+  } else if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  } else if (typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_VALUE('options', options);
   }
 
-  if (!callback && pos < arguments.length && arguments[pos] != null) {
-    throw new ERR_INVALID_ARG_VALUE('args', arguments[pos]);
+  if (callback && typeof callback !== 'function') {
+    throw new ERR_INVALID_ARG_VALUE('callback', callback);
   }
```

Comment:
```suggestion
  if (typeof options === 'function') {
    callback = options;
    options = null;
  } else {
    validateObject(options, 'options', { nullable: true });
    validateFunction(callback, 'callback');
  }

Reply:
This will need a similar workaround as with the other change. With this change, some arguments that previously could be omitted no longer can be.